api/src/Controller/Api/ProjectController.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\Document\Project;
+use App\Document\User;
+use App\Form\ProjectEditionType;
+use App\Security\ProjectVoter;
+use App\Utils\Validator;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Exception;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+#[Route('/api', name: 'api_')]
+class ProjectController extends AbstractController
+{
+    #[Route('/projects', name: 'get_all_projects', methods: 'GET')]
+    public function getAll(DocumentManager $dm): JsonResponse
+    {
+        /** @var User $user */
+        $user     = $this->getUser();
+        $projects = $dm->getRepository(Project::class)
+            ->findBy(['userId' => $user->getId()]);
+
+        return $this->json(['projects' => $projects], context: ['groups' => 'get_all_projects']);
+    }
+
+    #[Route('/projects/{id}', name: 'get_project', methods: 'GET')]
+    public function get(string $id, DocumentManager $dm): JsonResponse
+    {
+        try {
+            $project = $dm->getRepository(Project::class)->find($id);
+
+            if (!$project) {
+                throw $this->createNotFoundException('This project does not exist!');
+            }
+
+            $this->denyAccessUnlessGranted(ProjectVoter::VIEW, $project);
+        } catch (NotFoundHttpException $e) {
+            return $this->json(['errors' => [$e->getMessage()]], Response::HTTP_NOT_FOUND);
+        } catch (AccessDeniedException $e) {
+            return $this->json(['errors' => [$e->getMessage()]], Response::HTTP_UNAUTHORIZED);
+        } catch (Exception $e) {
+            return $this->json(['errors' => [$e->getMessage()]], Response::HTTP_BAD_REQUEST);
+        }
+
+        return $this->json([
+            'project' => $project,
+        ]);
```

Comment:
pour moi le le return `$this->json` il devrait etre dans le catch

Reply:
Si le dernier return provoque une erreur, le dernier catch va être trigger et pour moi ça serait pas une bad request mais une erreur logique la. Après je peux quand même le déplacer, j'ai pas la réponse absolue, je sais pas quelle est la best practice à ce sujet mais dans tous les cas ça ne changerai pas